src/azul/plugins/repository/canned/__init__.py

```diff
@@ -120,12 +123,47 @@ def list_sources(self,
         ]
 
     def _lookup_source_id(self, spec: SimpleSourceSpec) -> str:
-        return spec.name
+        return str(spec)
+
+    def parse_spec_name(self, spec_name: str) -> tuple[furl, Path, str]:
+        """
+        Parse the name of a source spec pointing at a canned staging area in a
+        GitHub repository.
```

Comment:
In my previous review I requested that you remove the awareness of this method of specs. You did that. But then you put that awareness right back into the name and the docstring. This method parses a URL to a directory in a GitHub repository at specific revision. The name and documentation should reflect only that. I'm not making these things up. They are engineering principles that every software engineer should be aware of and apply.

Read [this](https://en.wikipedia.org/wiki/Unix_philosophy) and [this](https://en.wikipedia.org/wiki/Separation_of_concerns#:~:text=In%20computer%20science%2C%20separation%20of,code%20of%20a%20computer%20program):